test/suites/smoke-test-dancelight/test-inbound-channel-validation-reception.ts

```diff
@@ -0,0 +1,95 @@
+import "@tanssi/api-augment/dancelight";
+
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import type { ApiPromise } from "@polkadot/api";
+import { findEraBlockUsingBinarySearch, PRIMARY_GOVERNANCE_CHANNEL_ID } from "../../utils";
+import { getBlockArray } from "@moonwall/util";
+import Bottleneck from "bottleneck";
+import type { FrameSystemEventRecord } from "@polkadot/types/lookup";
+
+const timePeriod = process.env.TIME_PERIOD ? Number(process.env.TIME_PERIOD) : 1 * 60 * 60 * 1000;
+
+describeSuite({
+    id: "SMOK10",
+    title: "Inbound channel validator reception",
+    foundationMethods: "read_only",
+    testCases: ({ it, context, log }) => {
+        let api: ApiPromise;
+        let blocksData: { blockNum: number; events: FrameSystemEventRecord[] }[];
+
+        beforeAll(async () => {
+            api = context.polkadotJs();
+
+            const blockNumArray = await getBlockArray(api, timePeriod);
+
+            const limiter = new Bottleneck({ maxConcurrent: 5, minTime: 100 });
+
+            const start = performance.now();
+            blocksData = await Promise.all(blockNumArray.map((num) => limiter.schedule(() => getBlockData(api, num))));
+            const end = performance.now();
+
+            log(`Blocks data fetching took: ${(end - start).toFixed(2)} ms. Fetched: ${blocksData.length} blocks.`);
+        });
+
+        it({
+            id: "C01",
+            title: "External index is correctly updated",
+            test: async () => {
+                const bondedEras = (await api.query.externalValidatorSlashes.bondedEras()).toJSON() as [
+                    [number, number, number],
+                ];
+                // Let's check 2 recent eras
+                for (const bondedEra of bondedEras.slice(-2)) {
+                    const result = await findEraBlockUsingBinarySearch(api, bondedEra[0]);
```

Comment:
This looks slow, is this the bottleneck that prevents us from testing more than 2 items? If so, maybe we could be smarter about it

Reply:
No, the ticket requirement is to test at least 2 latest bonded eras. I think we can easily go more, maybe 10.

Reply:
Ok, nevermind then